camel-core/src/main/java/org/apache/camel/impl/transformer/ProcessorTransformer.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.impl.transformer;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+import org.apache.camel.ExchangePattern;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+import org.apache.camel.impl.DefaultExchange;
+import org.apache.camel.model.transformer.EndpointTransformerDefinition;
+import org.apache.camel.processor.SendProcessor;
+import org.apache.camel.spi.DataType;
+import org.apache.camel.spi.Transformer;
+import org.apache.camel.support.ServiceSupport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A <a href="http://camel.apache.org/transformer.html">Transformer</a>
+ * leverages Processor to perform transformation.
+ */
+public class ProcessorTransformer extends Transformer {
+    private static final Logger LOG = LoggerFactory.getLogger(ProcessorTransformer.class);
+
+    private Processor processor;
+    private String transformerString;
+
+    public ProcessorTransformer(CamelContext context) {
+        setCamelContext(context);
+    }
+
+    /**
+     * Perform data transformation with specified from/to type using Processor.
+     * @param message message to apply transformation
+     * @param from 'from' data type
+     * @param to 'to' data type
+     */
+    @Override
+    public void transform(Message message, DataType from, DataType to) throws Exception {
+        Exchange exchange = message.getExchange();
+        CamelContext context = exchange.getContext();
+        if (from.isJavaType()) {
+            Object input = message.getBody();
+            Class<?> fromClass = context.getClassResolver().resolveClass(from.getName());
+            if (!fromClass.isAssignableFrom(input.getClass())) {
+                LOG.debug("Converting to '{}'", fromClass.getName());
+                input = context.getTypeConverter().mandatoryConvertTo(fromClass, input);
+                message.setBody(input);
+            }
+        }
+        
+        LOG.debug("Sending to transform processor '{}'", processor);
+        DefaultExchange transformExchange = new DefaultExchange(exchange);
```

Comment:
Not sure if we need to create a new exchange, we can send the existing exchange.